pkg/security/ImageScanService.go

```diff
@@ -479,6 +495,17 @@ func (impl ImageScanServiceImpl) FetchMinScanResultByAppIdAndEnvId(request *Imag
 				lowCount = lowCount + 1
 			}
 		}
+		if len(imageScanResult) > 0 {
+			scantoolId = imageScanResult[0].ScanToolId
+		} else {
+			scanToolHistoryMappings, err := impl.scanToolExecutionHistoryRepository.GetAllScanHistoriesByExecutionHistoryIds(scanExecutionIds)
+			if err != nil {
+				impl.Logger.Errorw("error in getting scanToolHistoryMappings", "err", err)
```

Comment:
same comment as above. We can extract the logic (something like getScanToolIdForReasonX) to remove duplicacy